config/v1/types_infrastructure.go

```diff
@@ -528,18 +528,23 @@ type AWSPlatformStatus struct {
 
 // AWSResourceTag is a tag to apply to AWS resources created for the cluster.
 type AWSResourceTag struct {
-	// key is the key of the tag
+	// key sets the key of the AWS resource tag key-value pair. Key is required when defining an AWS resource tag.
+	// Minimum length of 1. Maximum length of 128.
+	// The string can contain only the set of alphanumeric characters, space (' '), '_', '.', '/', '=', '+', '-', ':', '@' and cannot be prefixed with 'aws:'.
```

Comment:
For consistency with other APIs and how they are worded, I would suggest
```suggestion
	// Key should consist of between 1 and 128 characters, and may
	// contain only the set of alphanumeric characters, space (' '), '_', '.', '/', '=', '+', '-', ':', and '@'.
	// Key must not be start with 'aws:'.
```